src/nvim/mbyte.c

```diff
@@ -2461,12 +1967,9 @@ int mb_fix_col(int col, int row)
   col = check_col(col);
   row = check_row(row);
   if (has_mbyte && ScreenLines != NULL && col > 0
-      && ((enc_dbcs
-          && ScreenLines[LineOffset[row] + col] != NUL
-          && dbcs_screen_head_off(ScreenLines + LineOffset[row],
-            ScreenLines + LineOffset[row] + col))
-        || (enc_utf8 && ScreenLines[LineOffset[row] + col] == 0)))
+      && (enc_utf8 && ScreenLines[LineOffset[row] + col] == 0)) {
```

Comment:
This condition could be further simplified (`has_byte ` and `enc_utf8` are `true`).